accepted/00000-openapi.md

```diff
@@ -0,0 +1,184 @@
+        responses = {
+            @ApiResponse(
+                responseCode = "200", description = "Successful response",
+                content = @Content(
+                    mediaType = "application/json",
+                    schema = @Schema(
+                        type = "object",
+                        properties = {
+                            @StringToClassMapItem(key = "label", value = String.class),
+                            @StringToClassMapItem(key = "name", value = String.class),
+                            @StringToClassMapItem(key = "description", value = String.class)
+                        }
+                    )
+                )
+	        )
+	    },
+	    parameters = {
+            @Parameter(
+                name="projectLabel",
+                schema = @Schema(type = "string"),
+                required = true,
+                description = "Content Project label",
+                in = ParameterIn.QUERY
+            ),
+            @Parameter(cr
+                name="envLabel",
+                schema = @Schema(type = "string"),
+                required = true,
+                description = "Environment label",
+                in = ParameterIn.QUERY
+            )
+	    },
+        requestBody = @RequestBody(
+            description = "Request body description",
+            content = @Content(
+                mediaType = "application/json",
+                schema = @Schema(
+                    type = "object",
+                    properties = {
+                        @StringToClassMapItem(key = "name", value = String.class),
+                        @StringToClassMapItem(key = "description", value = String.class)
+                    }
+                )
+            )
+        )
+	)
+
+```
+
+##### A.1. Library Dependencies
+Two additional Maven dependencies will be added to Uyuni to support Swagger annotations:
+ - `io.swagger.core.v3.swagger-core and`
+ - `io.swagger.core.v3.swagger-annotations`
+
+##### A.2. Existing API Endpoints
+To facilitate the transition, a one-time scanner tool will be developed to automate the conversion of existing Javadoc-based documentation into Swagger annotations. This tool will be implemented as a Java Doclet that systematically scans the codebase to identify predefined patterns in the Javadoc comments. Once identified, the tool will annotate the corresponding methods with the appropriate `@Operation` annotations.
+
+Additionally, the scanner will detect the use of API serializers that describe the return types of endpoints. It will parse the associated Javadoc comments of these serializers to extract the necessary details and incorporate them into the `@Operation` definitions.
+
+##### A.3. XML-RPC API code examples
+
+For XML-RPC endpoints, it is possible to provide a specific section in the web UI containing sample scripts. In addition to this, the `description` and `examples` fields of Swagger operation documentation can be used to aid understanding.
```

Comment:
Can openAPI auto-generate code snippets as usage examples in any way?

How do we differentiate between XMLRPC and HTTP in the docs UI?